### PR TITLE
Feature/sqlrs workspace conventions

### PR DESCRIPTION
**Pull request by evilguest**
# Summary
Implement file-based `sqlrs` aliases across the local CLI workflow and align the surrounding docs, examples, tests, and release E2E tooling with that model.

This branch turns `prepare`/`plan`/`run` alias mode into the primary local UX, supports mixed alias/raw `prepare ... run` flows, fixes path-resolution semantics around alias files, and updates release happy-path validation to exercise repo-tracked aliases instead of ad-hoc raw prepare arguments.

## Changes

- documented the M2 local DX direction and alias-based workflow in roadmap, architecture docs, user guides, and ADRs
- introduced file-based prepare/run alias design and recorded follow-up decisions for:
  - mixed alias/raw composite grammar
  - alias path resolution bases
  - release/e2e usage expectations
- implemented CLI alias mode for:
  - `sqlrs plan <ref>`
  - `sqlrs prepare <ref>`
  - `sqlrs run <ref> --instance ...`
- added support for mixed composite execution where `prepare ... run ...` can combine alias and raw stages in any supported combination
- switched alias resolution to the approved contract:
  - alias refs resolve from the current working directory
  - file references inside alias files resolve relative to the alias file itself
- added repo-tracked example aliases for:
  - `chinook`
  - `sakila`
  - `flights`
  - `flights-multi-step`
  - `liquibase/jhipster-sample-app`
- updated example docs and example wrapper scripts to work correctly with alias-based execution from nested directories
- invalidated stale cached psql prepare states after the include-path fix by bumping the backend psql task hash schema
- moved release-local happy-path scenarios and cache-pressure scenario definitions to `prepareAlias`-based execution
- updated release E2E staging logic to copy alias files into temp workspaces and execute `prepare <alias>` instead of raw `prepare:*` commands
- added/updated extensive Go and Node test coverage around:
  - alias parsing/loading/dispatch
  - composite prepare/run grammar
  - alias path rebasing
  - init/init_wsl/status/prepare edge cases
  - release-local workflow and E2E runner behavior
- polished a few adjacent DX issues discovered during the branch:
  - widened `ls --states` budget for `PREPARE_ARGS` on wide TTYs
  - replaced `ping` with `timeout` in Windows daemon test helpers to make transient console windows less surprising

## Testing

- [x] `frontend/main` tests
- [ ] `backend/*` tests
- [x] Manual testing (optional)

Automated checks run during this branch:

- [x] `go test .\frontend\cli-go\... --cover`
- [x] `go test .\frontend\cli-go\internal\app`
- [x] `go test .\frontend\cli-go\internal\cli`
- [x] `node scripts/e2e-release/run-scenario.test.mjs`
- [x] `node scripts/e2e-release/run-cache-pressure-scenario.test.mjs`
- [x] `node scripts/e2e-release/release-local-workflow.test.mjs`

Coverage noted during branch validation:

- [x] `github.com/sqlrs/cli/internal/app` -> `96.8%`
- [x] `github.com/sqlrs/cli/internal/cli` -> `95.6%`

Manual checks run during this branch:

- [x] `examples\chinook> ..\..\dist\bin\sqlrs prepare ../chinook run:psql -f .\queries.sql`
- [x] spot-check of `sqlrs ls --states` output on a wide terminal
- [x] local verification that alias-backed example flows resolve paths correctly from nested working directories

## Related issues

- N/A

### CODE REVIEW

File: frontend/cli-go/internal/app/run.go [hunk 1]
```diff
@@ -234,3 +246,114 @@ func buildFileStep(shared []string, value string, workspaceRoot string, cwd stri
...
+		return "", nil, ExitErrorf(2, "Missing value for --file")
+	}
+	if path == "-" || path == pgbenchStdinPath {
+		return pgbenchStdinPath + weightSuffix, &pgbenchFileSource{UsesStdin: true}, nil
+	}
+	normalizedPath, _, err := normalizeFilePath(path, workspaceRoot, cwd, nil)
+	if err != nil {
+		return "", nil, err
+	}
+	return pgbenchStdinPath + weightSuffix, &pgbenchFileSource{Path: normalizedPath}, nil
```
[the remark this answers is not in the document]
Comment (evilguest) [reply frontend/cli-go/internal/app/run.go:332]: Won't fix. We're not allowing the commands that depend on the non-publically-visible scripts
